### PR TITLE
Fix Soft credit personal note ton limit to 255 characters (DB limit).

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Fix Soft credit personal note ton limit to 255 characters (DB limit).

Alternate to #12044 - replication details are the same & max impact is the same. As I needed to go into the code to do the fix proposed on the other PR I also fixed all the deprecation notices I hit at the same time (although not the one e-notice at the very end)

@KarinG can you please test & review this - it solves the same problem as your PR but per comments on that PR it uses the approach that we suggested on your PR

### CODE REVIEW

File: CRM/PCP/BAO/PCP.php [hunk 1]
```diff
@@ -453,7 +453,7 @@ public static function buildPcp($pcpId, &$page, &$elements = NULL) {
       $page->_defaults['pcp_is_anonymous'] = 0;
 
       $page->add('text', 'pcp_roll_nickname', ts('Name'), array('maxlength' => 30));
```
Comment (monishdeb) [inline CRM/PCP/BAO/PCP.php]: as @KarinG indicated, this also need to be changed to ```$page->addField('pcp_roll_nickname', ['entity' => 'ContributionSoft', 'context' => 'create']);```

Comment (eileenmcnaughton) [reply CRM/PCP/BAO/PCP.php]: Yep - I don't want to scope-creep this to other fields

**Comment by eileenmcnaughton** [comment]
Note this requires @KarinG to confirm she has tested it

**Comment by KarinG** [comment]
I'm sorry but I get cannot apply patch on this PR/8 files (and the prerequisite XML PR/1 file) to the 4.7.x site that I have PCP Events in production on - or I would roll this out to our sites right now. So all I could do is do exactly what you have already done - on a brand new vanilla civibuild 5.x - I don't need to replicate that; Time is very limited at the moment with CiviCamp Calgary only weeks away now.

**Comment by eileenmcnaughton** [comment]
@KarinG - OK well we need someone to replicate before we can merge somewhere - that's our requirement. If you can't do it then it can wait until someone does want to put the time into doing that. 

On a vanilla 5.x is fine but it does need to go through the review hoops

**Comment by eileenmcnaughton** [comment]
@KarinG why don't you do it at the sprint - it's probably a twenty minute job to finish review at this point & the sprint might be a good time

**Comment by monishdeb** [comment]
I have tested the patch for PCP ```pcp_personal_note``` field and as now I can fill more then 255 characters. Also in addition the changes made to use ```CRM_Core_Pseudoconstant::key()``` instead of ```CRM_Core_OptionGroup::getValue``` doesn't result into any indented regression and successfully get rid of notices. 

Hence merging this PR.

**Comment by eileenmcnaughton** [comment]
Thanks @monishdeb I know you are busy and this did not affect any of your customers but I appreciate getting it wrapped up

**Pull request merged**